fenix/app/src/test/java/org/mozilla/fenix/debugsettings/DefaultDebugSettingsCacheTest.kt

```diff
@@ -0,0 +1,54 @@
+package org.mozilla.fenix.debugsettings
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import kotlinx.coroutines.flow.drop
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.test.runTest
+import mozilla.components.support.test.robolectric.testContext
+import org.junit.After
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mozilla.fenix.debugsettings.data.DefaultDebugSettingsCache
+
+@RunWith(AndroidJUnit4::class)
+class DefaultDebugSettingsCacheTest {
+
+    @After
+    fun cleanUp() = runTest {
+        val defaultDebugSettingsCache = DefaultDebugSettingsCache(
+            context = testContext,
+            writeScope = this,
+        )
+        defaultDebugSettingsCache.clearPreferences()
+    }
+
+    @Test
+    fun `GIVEN the debug drawer is disabled WHEN the flag is made true THEN the store should have the correct value`() = runTest {
+        val flowEmissionsToIgnore = 2
+        val defaultDebugSettingsCache = DefaultDebugSettingsCache(
+            context = testContext,
+            writeScope = this,
+        )
```

Comment:
I'm not sure I'm following. I had some issues specific to data store + flows + coroutines that ultimately led me to the current version, but I'm certainly open to iterating on it.

Reply:
I assumed the issue here was that you wanted to use the `TestScope` provided by `runTest` as `this` in each test. The example above is how you can do that in the general case rather than in each test individually.

Reply:
Let's sync up on this over a call sometime so I can better explain the issues I was having. Very likely, using the [Turbine library](https://github.com/mozilla-mobile/firefox-android/pull/4829#discussion_r1426859685) will help clean all of this up.